addons/hr_holidays/models/hr_employee.py

```diff
@@ -168,3 +168,21 @@ def write(self, values):
             allocations = self.env['hr.leave.allocation'].sudo().search([('state', 'in', ['draft', 'confirm']), ('employee_id', 'in', self.ids)])
             allocations.write(hr_vals)
         return res
+
+    def action_time_off_analysis(self):
+        domain = [('holiday_type', '=', 'employee')]
+
+        if self.env.context.get('active_ids'):
+            domain.append(('employee_id', 'in', self.env.context.get('active_ids')))
+
+        return {
+            'name': _('Time Off Analysis'),
+            'type': 'ir.actions.act_window',
+            'res_model': 'hr.leave.report',
+            'view_mode': 'tree,form,pivot',
```

Comment:
Don't forget the search view id

```suggestion
            'view_mode': 'tree,form,pivot',
            'search_view_id': self.env.ref('view_hr_holidays_filter_report').id,
```